climada/util/api_client.py

```diff
@@ -654,32 +630,29 @@ def get_exposures(self, exposures_type, dump_dir=SYSTEM_DIR, max_datasets=10, **
             number, a ValueError is raised. Setting it to 0 or a negative number inactivates the
             limit. Default is 10.
         **kwargs :
-            additional parameters passed on to `Client.get_datasets`
+            additional parameters passed on to `Client.list_dataset_infos`
 
         Returns
         -------
         climada.entity.exposures.Exposures
             The combined exposures object
         """
         if 'data_type' in kwargs:
-            raise ValueError("data_type is already given as hazard_type")
+            raise ValueError("data_type is already given as exposures_type")
         if not exposures_type in EXP_TYPES:
             raise ValueError("Valid exposures types are a subset of CLIMADA exposures types."
                              f" Currently these types are supported: {EXP_TYPES}")
-        datasets = self.get_datasets(data_type=exposures_type, **kwargs)
-
-        self._check_datasets_for_concatenation(datasets, max_datasets)
-
-        return self.to_exposures(datasets, dump_dir)
+        dataset = self.get_dataset_info(data_type=exposures_type, **kwargs)
+        return self.to_exposures(dataset, dump_dir)
 
-    def to_exposures(self, datasets, dump_dir=SYSTEM_DIR):
+    def to_exposures(self, dataset, dump_dir=SYSTEM_DIR):
```

Comment:
Same question as for `to_hazard`. Should this be private?

Reply:
s.a.